### ISSUE TITLE
Wrong unused-import warning when redefining import name inside class

### ISSUE THREAD

**Issue by federicobond** [PyCQA/pylint#900]
The following code:

``` python
import foo

class bar(object):
    foo = foo.baz
```

Triggers this pylint warning:

```
W:  1, 0: Unused import foo (unused-import)
```

But should not, because the `foo` module import is used to build `bar.foo`.

**Comment by moylop260** [comment]
`pylint --version`

**Comment by PCManticore** [comment]
I think this is a duplicate of https://github.com/PyCQA/pylint/issues/848. The gist of it is that Pylint becomes confused by your foo reassignment.

**Comment by federicobond** [comment]
````
pylint 1.5.1, 
astroid 1.4.1
````

You are right, it's a duplicate of the issue you mention.

**Comment by PCManticore** [comment]
Fixed in master branch.
